src/modules/Dashboard/features/RespondentData/RespondentDataSummary/RespondentDataSummary.tsx

```diff
@@ -1,33 +1,106 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
 
-import { Activity } from 'redux/modules';
-import { useBreadcrumbs } from 'shared/hooks';
-import { StyledContainer } from 'shared/styles';
+import { useAsync, useBreadcrumbs } from 'shared/hooks';
+import { Spinner, Svg } from 'shared/components';
+import { useDecryptedIdentifiers } from 'modules/Dashboard/hooks';
+import { StyledContainer, StyledTitleLarge, variables } from 'shared/styles';
+import {
+  DatavizActivity,
+  Version,
+  getIdentifiersApi,
+  getSummaryActivitiesApi,
+  getVersionsApi,
+} from 'api';
 
-import { mockedActivities } from '../mock';
-import { Report } from './Report';
 import { ReportMenu } from './ReportMenu';
+import { Report } from './Report';
+import { StyledEmptyContainer, StyledEmptyReview } from './RespondentDataSummary.styles';
+import { Identifier } from './RespondentDataSummary.types';
 
 export const RespondentDataSummary = () => {
   const { t } = useTranslation();
-  const activities = mockedActivities;
-  const [selectedActivity, setSelectedActivity] = useState<Activity>(activities[0]);
+  const { appletId, respondentId } = useParams();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [activities, setActivities] = useState<DatavizActivity[]>([]);
+  const [selectedActivity, setSelectedActivity] = useState<DatavizActivity>();
+  const [versions, setVersions] = useState<Version[]>([]);
+  const [identifiers, setIdentifiers] = useState<Identifier[]>([]);
+
+  const getDecryptedIdentifiers = useDecryptedIdentifiers();
+
+  const { execute: getSummaryActivities } = useAsync(getSummaryActivitiesApi);
+  const { execute: getIdentifiers } = useAsync(getIdentifiersApi);
+  const { execute: getVersions } = useAsync(getVersionsApi);
+
   useBreadcrumbs([
     {
       icon: 'chart',
       label: t('summary'),
     },
   ]);
 
+  useEffect(() => {
+    const fetchActivities = async () => {
+      if (!appletId || !respondentId) return;
+      try {
+        const result = await getSummaryActivities({
+          appletId,
+        });
+        setActivities(result.data?.result);
+      } finally {
+        setIsLoading(false);
```

Comment:
we have loading and value in useAsync

Reply:
when using loading from useAsync, the screen with empty state blinks at the beginning of loading

Reply:
what about value? and mb when initial state be null it will be fixed?

Reply:
it's not because of the value in the state
it appears due to the isLoading from useAsync at the time of the start of rendering is false